examples_test.go

```diff
@@ -0,0 +1,166 @@
+package altsrc_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	altsrc "github.com/urfave/cli-altsrc/v3"
+	"github.com/urfave/cli/v3"
+)
+
+var (
+	testdataDir string
+)
+
+func init() {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	if err := setup(ctx); err != nil {
+		panic(err)
+	}
+}
+
+func setup(ctx context.Context) error {
+	topBytes, err := exec.CommandContext(ctx, "git", "rev-parse", "--show-toplevel").Output()
+	if err != nil {
+		return err
+	}
+
+	testdataDir = filepath.Join(strings.TrimSpace(string(topBytes)), ".testdata")
+	return nil
+}
+
+func ExampleYAML() {
```

Comment:
You can collapse the code in these 3 examples with a 

setupApp(ValueSources..) convenience method. 

That way all the examples have to do is pass in the value sources and rest remains the same.

Reply:
I was hesitant to abstract too much of it because the bodies of these will appear in `godoc` documentation but without the setup code or abstracted/shared code 🤔